src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -47,11 +52,15 @@ bool emberAfGeneralCommissioningClusterCommissioningCompleteCallback(EndpointId
 }
 
 bool emberAfGeneralCommissioningClusterSetRegulatoryConfigCallback(EndpointId endpoint, app::CommandHandler * commandObj,
-                                                                   uint8_t location, uint8_t * countryCode, uint64_t breadcrumb,
-                                                                   uint32_t timeoutMs)
+                                                                   uint8_t location, chip::ByteSpan countryCode,
+                                                                   uint64_t breadcrumb, uint32_t timeoutMs)
 {
+    char countryCodePtr[kMaxCountryCodeLen + 1];
+    Platform::CopyString(countryCodePtr, sizeof(countryCodePtr), countryCode);
+    countryCodePtr[kMaxCountryCodeLen] = '\0';
+
     CHIP_ERROR err = DeviceLayer::Internal::DeviceControlServer::DeviceControlSvr().SetRegulatoryConfig(
-        location, reinterpret_cast<const char *>(countryCode), breadcrumb);
+        location, reinterpret_cast<const char *>(countryCodePtr), breadcrumb);
```

Comment:
This cast is not needed.  But also, followup issue to change SetRegulatoryConfig to not require a null-terminated string, please.